paddle/operators/while_op.cc

```diff
@@ -205,6 +208,8 @@ class WhileGradOp : public framework::OperatorBase {
         sum_op->Run(cur_scope, dev_place);
         cur_scope.Rename(new_inside_name, inside_grad_name);
       }
+      dev_ctx.Wait();
```

Comment:
what is the cost of adding this `dev_ctx.Wait();`?

Reply:
I have not make detailed test yet. But if we do not delete the step scope, we can not training a larger RNN model because of OutOfMemory.